### PR TITLE
Create Bootstrap sample

### PR DESCRIPTION
**Pull request by NenoStefanov**

### CODE REVIEW

File: BootstrapSample/README.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+}
+```
+
+### Bootstrap Code Generation 
+
+All of the work is done at [`CodeGenerationFragment.java`](src/main/java/com/miracl/mpinsdk/bootstrapsample/CodeGenerationFragment.java), which is presented when `Code Generation` tab is clicked.
+
+When you run the demo there are two buttons on the screen:
+* `Select User` - opens a list of already registered identities from which user can select;
+* `Register New User` - uses the registration flow described [here](https://github.com/miracl/sample-mobile-app-android/blob/master/MobileAppLoginSample/README.md#registeruseractivityjava) to create a new identity from scratch.
```
Comment (tatiNo5) [inline BootstrapSample/README.md]: from scratch but skipping the verification step.

File: DvsSample/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -32,8 +32,8 @@ Please, press the button to proceed.</string>
 
     <!-- M-Pin configuration -->
     <string name="mpin_backend" formatted="false" translatable="false">https://api.mpin.io</string>
-    <string name="mpin_cid" formatted="false" translatable="false"></string>
+    <string name="mpin_cid" formatted="false" translatable="false">18ed2c66-0955-4617-8abe-c42a0846fa2f</string>
 
     <!-- Custom service configuration -->
-    <string name="access_code_service_base_url" formatted="false" translatable="false"></string>
+    <string name="access_code_service_base_url" formatted="false" translatable="false">http://192.168.0.101:5001</string>
```
Comment (tatiNo5) [inline DvsSample/src/main/res/values/strings.xml]: Should this be committed?

File: BootstrapSample/README.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+All of the work is done at [`CodeGenerationFragment.java`](src/main/java/com/miracl/mpinsdk/bootstrapsample/CodeGenerationFragment.java), which is presented when `Code Generation` tab is clicked.
+
+When you run the demo there are two buttons on the screen:
+* `Select User` - opens a list of already registered identities from which user can select;
+* `Register New User` - uses the registration flow described [here](https://github.com/miracl/sample-mobile-app-android/blob/master/MobileAppLoginSample/README.md#registeruseractivityjava) to create a new identity from scratch in order to have an identity to generate the bootstrap code with.
+
+Choosing `Select User` and selecting an already registered identity from the list displays the following information:
+
+<img src="images/selected-user.png" width="266">
```
Comment (Foriger) [inline BootstrapSample/README.md]: On the picture, there are two MPin Backend labels/

File: BootstrapSample/README.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+If you want to move the selected user identity to another device, press `Generate Code` button. Then [getMfaSdk().startAuthenticationRegCode](https://github.com/miracl/mfa-client-sdk-android#status-startauthenticationregcodeuser-user) method is called which kicks-off the authentication process:
+
+```
+SampleApplication.getMfaSdk().startAuthenticationRegCode(mCurrentUser, new MPinMfaAsync.Callback<Void>() {
+    ...
+}
+```
+
+If status OK is returned, the user is asked for their PIN. Then [getMfaSdk().finishAuthenticationRegCode](https://github.com/miracl/mfa-client-sdk-android#status-finishauthenticationregcodeuser-user-string-multifactor-regcode-regcode) method is called to finalize the authentication process:
```
Comment (Foriger) [inline BootstrapSample/README.md]: If status `OK` is returned, ....

File: BootstrapSample/README.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+```
+SampleApplication.getMfaSdk().startRegistration(SampleApplication.getCurrentAccessCode(), mCurrentUser, null, bootstrapCode, new MPinMfaAsync.Callback<Void>() {
+    ...
+}
+```
+
+If status OK is returned, the `mEnterPinDialog` is shown to the user to create a PIN code for their new identity on this device: 
+
+<img src="images/enter-pin-registration-with-code.png" width="266">
```
Comment (Foriger) [inline BootstrapSample/README.md]: Is there a keyboard here, or button? 
How this button is submitted?

File: BootstrapSample/README.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        }
+    }
+}
+```
+
+and a message is presented to show the user they have already a registered identity on that device:
+
+<img src="images/registration-ok-reg-code.png" width="266">
+
+### AccessCodeObtainingTask.java
```
[the remark this answers is not in the document]
Comment (NenoStefanov) [reply BootstrapSample/README.md]: The section explains how we get an access code that is used to start the registration.

File: BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/AccessCodeObtainingTask.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        AccessCodeServiceApi accessCodeServiceApi = retrofit.create(AccessCodeServiceApi.class);
+
+        try {
+            // Get the auth url from a demo service
+            Response<AuthorizeUrlInfo> responseAuthUrl = accessCodeServiceApi.getAuthURL().execute();
+            if (responseAuthUrl.code() == HTTP_CODE_OK) {
+                AuthorizeUrlInfo urlInfo = responseAuthUrl.body();
+
+                StringBuilder accessCodeContainer = new StringBuilder();
+                // Use the auth url in order to receive and access code
```
Comment (tatiNo5) [inline BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/AccessCodeObtainingTask.java]: ... to receive **an** access code

File: BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                    // Set the cid and the backend with which the SDK will be configured
+                    sdk.setCid(getString(R.string.mpin_cid));
+                    Status setBackendStatus = sdk.setBackend(getString(R.string.mpin_backend));
+                    if (setBackendStatus.getStatusCode() != com.miracl.mpinsdk.model.Status.Code.OK) {
+                        runOnUiThread(new Runnable() {
+
+                            @Override
+                            public void run() {
+                                Toast.makeText(MainActivity.this,
+                                        "The MPin SDK did not initialize properly. Check you backend and CID configuration",
```
Comment (tatiNo5) [inline BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/MainActivity.java]: Check **your** backend...

File: BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/RegisterUserActivity.java [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+                mCurrentUser = result;
+                // After we have a user, we can start the registration process for it. If successful this will trigger sending a
+                // confirmation email from the current backend
+                SampleApplication.getMfaSdk()
+                        .startRegistration(SampleApplication.getCurrentAccessCode(), mCurrentUser, new MPinMfaAsync.Callback<Void>() {
+
+                            @Override
+                            protected void onSuccess(@Nullable Void result) {
+                                // When the registration process is started successfully for a user, it is stored in the SDK and is
+                                // associated with the current backend at the time.
```
Comment (tatiNo5) [inline BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/RegisterUserActivity.java]: maybe .....backend at **that** time ?

Comment (tatiNo5) [reply BootstrapSample/src/main/java/com/miracl/mpinsdk/bootstrapsample/RegisterUserActivity.java]: When the registration process is started successfully for a user, the identity is stored in the SDK, associating it with the current set backend.

**Pull request merged**